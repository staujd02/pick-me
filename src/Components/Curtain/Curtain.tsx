import React from 'react';
import Slide from '@material-ui/core/Slide';

class Curtain extends React.Component<CurtainProps, CurtainState> {
    render() {
        const { open } = this.props;
        return (
            <Slide direction={'down'} in={open}>
                <div id="cycler-controls">
                    {this.props.children}
                </div>
            </Slide>);
    }
}

export default Curtain;