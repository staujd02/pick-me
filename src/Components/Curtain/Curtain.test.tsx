import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import Curtain from './Curtain';

describe("The Curtain", () => {

    const children = <div>Now you see me!</div>;
    let wrapper: ReactWrapper<CurtainProps, CurtainState>;

    beforeEach(() => {
        wrapper = mount(
            <Curtain open={true}>
                {children}
            </Curtain>);
    });

    it('renders correctly', () => expect(wrapper.html()).toMatchSnapshot());
});