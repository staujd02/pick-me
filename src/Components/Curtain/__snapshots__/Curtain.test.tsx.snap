// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Curtain renders correctly 1`] = `"<div id=\\"cycler-controls\\" style=\\"webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;\\"><div>Now you see me!</div></div>"`;
